### DISCUSSION TITLE
Color does not work for hits of type box

### DISCUSSION THREAD

**Discussion by Lysarina** [HSF/phoenix#663] [Q&A]
Hi!
I'm working with setting up Phoenix for the LDMX experiment. I am trying to plot both electron clusters and hits, and the idea is to have hits in the same cluster drawn in the same color. However, I've run into the problem that setting a color for a hit object of type box in a JSON event file does not work, even if this is supposed to be possible according to the event data format page in the developer's guide. For example, here the color attribute is set to 0xff00ff (fuchsia) for all hits in the collection ecalClusters, but the boxes still appear red. This seems to only be a problem with box hits, as setting e.g. tracks to the same color works just fine, as seen in the picture.
![image](https://github.com/HSF/phoenix/assets/62296952/f3404556-420c-4b00-bf0f-bc2b6e895e11)
Here is a couple of hit objects in my JSON file:
![image](https://github.com/HSF/phoenix/assets/62296952/dd96cbd2-9db9-4d50-9780-d857d932690d)
It is possible to change the hit color in the Phoenix menu, so it is possible for boxes to have a different color, just not from a JSON file -- but I would need this functionality, as I want to have different colors in the same collection. Even if I did have separate collections it isn't very feasible to manually change colors for a bunch of clusters.
The colors also seem to work with hits of type point, but I would like to have the option of size that the box type provides.
I have tried loading in the JSON file in my own Phoenix setup, in the Phoenix playground, as well as from the locally run yarn project pulled from the Phoenix GitHub, and the result is the same in all cases.
Could this be fixed?

**Comment by EdwardMoyse** [comment]
Hi @Lysarina - thanks for reporting this. Can you open an issue, and attach your json file to it?

**Reply by Lysarina** [reply]
Done!

I dug around and think I found the source of the problem as well, as I wrote in the issue. I noticed that the color for a collection of hits is set as the color of the first hit -- so I'm guessing it's not possible to have several different colors in one collection?
